lib/controllers/root-controller.js

```diff
@@ -94,6 +94,16 @@ export default class RootController extends React.Component {
     this.subscription = new CompositeDisposable(
       this.props.repository.onMergeError(this.gitTabTracker.ensureVisible),
     );
+
+    this.props.commandRegistry.onDidDispatch(event => {
```

Comment:
thanks for taking this on, @vanessayuenn !

It makes me nervous that there are no unit tests for this code, although I agree with you that this is tricky to test.

What if we stubbed `this.props.commandRegistry.onWillDispatch` and added `event.detail[0].contextCommand`?  Would this allow us to simulate a context event from a unit test?  Of course, it would have to . be scoped to just this unit test to prevent leakage.

Or maybe @smashwilson and @kuychaco have ideas here?

Reply:
@vanessayuenn well done digging to find this `contextCommand` flag! Super handy! ✨ 

As far as how to unit test this, looks like there's an [undocumented] `detail` argument we can supply to the `dispatch` method on `commandRegistry` -- https://github.com/atom/atom/blob/v1.31.2/src/command-registry.js#L270

So it'd look something like this:

```
atom.commands.dispatch(
  '.github-FilePatchListView-path', 
  'github:discard-changes-in-selected-files', 
  {contextCommand: true}
)
```

We can test that we don't record events if `contextCommand` is not true, as well as if the event type does not start with `github:`.

One thing to note -- some of our context menu actions are bound to core commands (like `core:confirm`) so we wouldn't be catching these context menu actions. For example:

```
'.github-FilePatchView.is-staged': [
    {
      'label': 'Unstage Selection'
      'command': 'core:confirm'
    }
  ]
```

This is fine. Just something to keep in mind when we're looking at metrics, as well as when we're deciding what commands to bind context menu actions to.

Reply:
Thanks for the input & ideas!